### PR TITLE
Add support for vue single file components

### PR DESCRIPTION
**Pull request by Havunen**
relates to https://github.com/pahen/madge/issues/122

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -389,6 +390,37 @@ function commonJSLookup(options) {
   return result;
 }
 
+function vueLookup(options) {
+  const dependency = options.dependency;
+
+  if (!dependency) {
```
Comment (XhmikosR) [inline index.js:396]: This whole block is probably better written with a `switch` statement.

**Comment by Havunen** [comment]
@XhmikosR Shall we merge this

File: index.js (hunk 1 above)
Comment (Havunen) [reply index.js:396]: it cannot be original table lookup switch because those are endsWith statements, they require runtime evaluation

**Comment by Havunen** [comment]
@XhmikosR can we merge this ?

**Comment by XhmikosR** [comment]
Even if this is merged, I personally won't merge https://github.com/dependents/node-precinct/pull/124 due to the huge dependency: https://packagephobia.com/result?p=detective-vue2

**Pull request merged**